src/Raven.Server/ServerWide/Commands/Subscriptions/PutSubscriptionCommand.cs

```diff
@@ -52,6 +54,8 @@ public override unsafe void Execute(TransactionOperationContext context, Table i
             while (tryToSetName)
             {
                 var subscriptionItemName = SubscriptionState.GenerateSubscriptionItemKeyName(DatabaseName, SubscriptionName);
+                if (subscriptionItemName.Length > AbstractPager.MaxKeySize)
```

Comment:
I think it would be better to use the same constraint as for documents. Which is 512. Check here: https://github.com/ravendb/ravendb/pull/9014

That should be applied to initial subscription name only, not the generated one.

Reply:
The generated subscription name should never be too long, I think